meshmode/dof_array.py

```diff
@@ -223,6 +214,10 @@ def imag(self):
 
     def _ibop(self, f, arg):
         """Generic in-place binary operator without any broadcast support."""
+        from warnings import warn
+        warn("In-place operations on DOFArrays are deprecated. "
```

Comment:
I don't remember why in-place updates were added to the `DOFArray`. I feel like I was discussing with @mtcam at the time. As you can see, I'm inclined to get rid of them. @mtcam, do you remember?

Reply:
That should be my fault (added in inducer/meshmode#101), but I'm fine with deprecating it. I can work around it locally, if necessary.

My use case was that I was doing a lot of `x + bump_function` for each DOF in the mesh and it seemed like a good idea to do it in place.

Reply:
OK, sounds good. Thanks for chiming in @mtcam and @alexfikl! Let's stick to deprecating inplace ops for `DOFArray` then.